src/ribbon/buttonbar.cpp

```diff
@@ -406,29 +428,32 @@ wxRibbonButtonBarButtonBase* wxRibbonButtonBar::InsertButton(
                 size_t pos,
                 int button_id,
                 const wxString& label,
-                const wxBitmap& bitmap,
-                const wxBitmap& bitmap_small,
-                const wxBitmap& bitmap_disabled,
-                const wxBitmap& bitmap_small_disabled,
+                const wxBitmapBundle& bitmap,
+                const wxBitmapBundle& bitmap_small,
+                const wxBitmapBundle& bitmap_disabled,
+                const wxBitmapBundle& bitmap_small_disabled,
                 wxRibbonButtonKind kind,
                 const wxString& help_string)
 {
     wxASSERT(bitmap.IsOk() || bitmap_small.IsOk());
     if(m_buttons.IsEmpty())
     {
+        bool setLarge = m_bitmap_size_large.x == 0 || m_bitmap_size_large.y == 0;
+        bool setSmall = m_bitmap_size_small.x == 0 || m_bitmap_size_small.y == 0;
+
         if(bitmap.IsOk())
         {
-            m_bitmap_size_large = bitmap.GetLogicalSize();
-            if(!bitmap_small.IsOk())
+            if(setLarge)m_bitmap_size_large = bitmap.GetPreferredLogicalSizeFor(m_ribbonBar);
```

Comment:
```suggestion
            if ( setLarge )
                m_bitmap_size_large = bitmap.GetPreferredLogicalSizeFor(m_ribbonBar);
```
Please don't put compound statements all on the same line.